src/gui/folder.cpp

```diff
@@ -925,15 +925,30 @@ void Folder::slotItemCompleted(const SyncFileItemPtr &item)
         return;
     }
 
-    // add new directories or remove gone away dirs to the watcher
+#ifdef Q_OS_LINUX
```

Comment:
So mac and windows are already recursive?

Reply:
Yes, I double checked. Their addFolder/removeFolder functions are literally {} (and the elaborate follow up patch removes them and makes the linux folderwatcher recursive too).